addons/sale_coupon/tests/test_program_numbers.py

```diff
@@ -687,9 +688,9 @@ def test_coupon_and_coupon_discount_fixed_amount_tax_excl(self):
         # - a fixed amount (greater than remaining amount to pay) +
         # - discount amount
         # And user should be able to swap the promotion order with a meaningful result.
-        self.assertEqual(order.amount_tax, 13.5)
+        self.assertEqual(order.amount_tax, 0)
```

Comment:
> #TODO fix numbers
        # Need an in-depth inspection on the behavior with
        # - multiple product with different VAT +
        # - a fixed amount (greater than remaining amount to pay) +
        # - discount amount
        # And user should be able to swap the promotion order with a meaningful result.

Your fix is supposed to handle all that, now that you are tackling this issue that has been postponed for a long time.
I think every part of that comment is true, if we don't want to have a bad risk/reward ratio with this fix.

I actually wrote my review message before reading this part of the code which actually.. Explains the exact same thing it seems, making it even more relevant.

Reply:
> # - multiple product with different VAT +
This is the test that have been added

> # - a fixed amount (greater than remaining amount to pay) + discount amount
This is the test case that was already present (the one you reviewed on), it's the reason for the non-test change in the second commit

> # And user should be able to swap the promotion order with a meaningful result.
Currently promotion are applied in the order of their application, which might result in different SO (if e.g. they have a total discount value of > 100% it will always be the later who get reduced)
But i'll add a test to see if applying 2 coupon/promo in different order result in similar SO if their value < product value